src/module/actor/mixins/actor-conditions.js

```diff
@@ -50,11 +79,8 @@ export const ActorConditionsMixin = (superclass) => class extends superclass {
                         const promise = this.createEmbeddedDocuments("Item", [itemData]);
                         promise.then((createdItems) => {
                             if (createdItems && createdItems.length > 0) {
-                                const updateData = {};
-                                updateData[`data.conditions.${conditionName}`] = true;
-                                this.update(updateData).then(() => {
-                                    Hooks.callAll("onActorSetCondition", {actor: this, item: createdItems[0], conditionName: conditionName, enabled: enabled});
-                                });
+                                this._updateActor(conditionName, true);
+                                Hooks.callAll("onActorSetCondition", {actor: this, item: createdItems[0], conditionName: conditionName, enabled: enabled});
```

Comment:
Why did you take the onActorSetCondition callback out of the actor update? It was done on purpose so that the callback would not fire until the actor was fully finished updating.

Reply:
I think it got lost in the shuffle. You'll notice the update is calling a different method entirely. I'll make it a callback again.